client/core/trade.go

```diff
@@ -1239,14 +1244,111 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 				"but we have not seen the counter-party's redemption yet. This could just"+
 				" be network latency.", match)
 		}
+
 		match.setSwapConfirms(int64(confs))
+
 		t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
 		return false, false
 	}
 
 	return false, false
 }
 
+// updateFeesPaid starts a goroutine that updates the fees used for dynamic fee
+// checker transactions. We do not know the exact fees the tx will use until
+// they are mined, so this waits until they are mined and updates the value for
+// the entire trade.
+//
+// TODO: This method does not persist if the client is shut down. It probably
+// should because payed fees will be incorrect forever for this trade if not.
+//
+// This method accesses match fields and MUST be called with the trackedTrade
+// mutex lock held for reads.
```

Comment:
Looks like the goroutine it starts has it's own locking, while `updateFeesPaid` does not access match fields itself.

Reply:
It takes the trade wallets, but I guess those never change so doesn't need the read lock. Will remove comment.